core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1503,10 +1535,9 @@ public void reportLostFile(long fileId) throws FileDoesNotExistException {
    * @throws AccessControlException if permission checking fails
    */
   // TODO(jiri): Make it possible to load UFS objects recursively.
```

Comment:
I am not sure if we would want to ever load metadata recursively with unlimited depth ... your PR addresses the use cases this TODO was meant to address